src/workerd/io/actor-sqlite.c++

```diff
@@ -513,10 +523,21 @@ ActorCacheInterface::DeleteAllResults ActorSqlite::deleteAll(WriteOptions option
     }
   }
 
-  if (localAlarmState == kj::none && !deleteAllCommitScheduled) {
-    // If we're not going to perform a write to restore alarm state, we'll want to make sure the
-    // commit callback is called for the deleteAll().
-    commitTasks.add(outputGate.lockWhile(kj::evalLater([this]() mutable -> kj::Promise<void> {
+  // We want to ensure a commit is scheduled during the deleteAll() call, but if we end up doing a
+  // write to restore alarm state afterward, we can skip the original scheduled commit, since the
+  // write will queue its own commit.
+  kj::Maybe<kj::Own<bool>> maybeCanSkipDeleteAllCommit;
```

Comment:
I'm not sure if cancelling the scheduled commit is strictly necessary, or if there isn't a simpler way to do it that preserves the desired invariants in all cases (exceptions, cancellation, etc.).  But without the cancellation, I noticed via the tests that there was otherwise a redundant commit call.